src/firefly/js/tables/ui/TablePanelOptions.jsx

```diff
@@ -3,63 +3,23 @@
  */
 
 import React from 'react';
-import {isEmpty, cloneDeep} from 'lodash';
+import {isEmpty, cloneDeep, get} from 'lodash';
 
-import {BasicTable} from './BasicTable.jsx';
+import {BasicTableView} from './BasicTableView.jsx';
 import {SelectInfo} from '../SelectInfo.js';
+import {SortInfo, SORT_ASC} from '../SortInfo.js';
+import {sortTable} from '../TableUtil.js';
 import {InputField} from '../../ui/InputField.jsx';
 import {intValidator} from '../../util/Validate.js';
 
 
-function prepareOptionData(columns) {
-
-    var data = columns.map( (v, idx) => {
-        return [v.name];
-    } );
-    var cols = [{name: 'Column', visibility: 'show', prefWidth: 20}];
-    var selectInfoCls = SelectInfo.newInstance({});
-    selectInfoCls.data.rowCount = data.length;
-    columns.forEach( (v, idx) => {
-        selectInfoCls.setRowSelect(idx, v.visibility === 'show');
-    } );
-    var tableRowCount = data.length;
-
-    return {cols, data, tableRowCount, selectInfoCls };
-}
-
 export const TablePanelOptions = (props) => {
-    const {columns, pageSize, showUnits, showFilters, onChange} = props;
+    const {columns, pageSize, showUnits, showFilters, onChange, colSortDir} = props;
     if (isEmpty(columns)) return false;
```

Comment:
Is it OK to return a primitive from a functional component?
I saw this: https://github.com/facebook/react/issues/5355